docker-compose.yml

```diff
@@ -3,6 +3,30 @@ version: "3.8"
 # TODO: Containerise frontend as well
 
 services:
+  web:
+    build: ./frontend/charity-management-app
+    ports:
+      - 5173:5173
+    volumes: 
+      - ./frontend/charity-management-app:/app
+      - /app/node_modules
+    develop:
+      # specify the files to watch for changes
+      watch:
+        # it'll watch for changes in package.json and package-lock.json and rebuild the container and image if there are any changes.
+        # Basically if we add any new packages we should "rebuild", which essentially means we rebuild the image from the dockerfile.
+        # This is so node_modules can be reloaded and be up to date in the container
+        # This is rebuild basically runs `docker-compose up --rebuild` and so it rebuilds the image and
+        # resets the container
+        - path: ./frontend/charity-management-app/package.json
+          action: rebuild
```

Comment:
This looks good, but can we add a rebuild action for when the package-lock.json changes as well?